In [2]:
import numpy as np
import pandas as pd
!conda install -c conda-forge folium=0.5.0 --yes
import folium


Solving environment: done

## Package Plan ##

  environment location: /opt/conda/envs/Python36

  added / updated specs: 
    - folium=0.5.0


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    openssl-1.1.1g             |       h516909a_0         2.1 MB  conda-forge
    branca-0.4.1               |             py_0          26 KB  conda-forge
    ca-certificates-2020.4.5.1 |       hecc5488_0         146 KB  conda-forge
    python_abi-3.6             |          1_cp36m           4 KB  conda-forge
    altair-4.1.0               |             py_1         614 KB  conda-forge
    vincent-0.4.4              |             py_1          28 KB  conda-forge
    folium-0.5.0               |             py_0          45 KB  conda-forge
    certifi-2020.4.5.1         |   py36h9f0ad1d_0         151 KB  conda-forge
    ------------------------------------------------------------
                       

In [3]:
dfs = pd.read_html('https://en.wikipedia.org/wiki/List_of_postal_codes_of_Canada:_M')
dfs[0].tail()

Postal Code       Borough  \
175         M5Z  Not assigned   
176         M6Z  Not assigned   
177         M7Z  Not assigned   
178         M8Z     Etobicoke   
179         M9Z  Not assigned   

                                          Neighborhood  
175                                                NaN  
176                                                NaN  
177                                                NaN  
178  Mimico NW, The Queensway West, South of Bloor,...  
179                                                NaN

In [11]:
#convert list to dataframe
df = pd.DataFrame(dfs[0])

# Get names of indexes for which borough is not assigned
indexNames = df[ df['Borough'] == 'Not assigned'].index
# Delete these row indexes from dataFrame
df.drop(indexNames , inplace=True)

# where neighborhood is not assigned set it to equal value for borough
indexNames1 = df[ df['Neighborhood'] == 'Not assigned'].index
#print(indexNames1) --> no neighborhoods not assigned

df.shape


(103, 3)

Part 1.
Wikipedia table read with pandas and converted from list to dataframe. 
Rows where Borough was not assigned were deleted and there were no rows found where neighborhood was not assigned.
There are 103 rows and 3  coloumns.